packages/@uppy/transloadit/src/AssemblyOptions.js

```diff
@@ -62,9 +62,13 @@ class AssemblyOptions {
    * Get Assembly options for a file.
    */
   async #getAssemblyOptions (file) {
-    const options = this.opts
+    if (file == null) return undefined
 
+    const options = this.opts
     const assemblyOptions = await options.getAssemblyOptions(file, options)
+
+    if (file == null) return undefined
```

Comment:
Twice the check? Is that the race condition bug again, where file could be removed at any point?

Reply:
It could had been removed during the `await`, e.g. if the user hit cancel while we're waiting for the options IIUC.